Update CSI migration feature status, and remove docs for unsupported plugins (#35685)
kubernetes/website: content/en/docs/concepts/storage/persistent-volumes.md

feature:
  title: Storage orchestration
  description: >

>     Automatically mount the storage system of your choice, whether from local storage, a public cloud provider such as <a href="https://cloud.google.com/storage/">GCP</a> or <a href="https://aws.amazon.com/products/storage/">AWS</a>, or a network storage system such as NFS, iSCSI, Gluster, Ceph, Cinder.

Reviewer: We prefer to list in alphabetical order for third party services and products.. Since we're touching this line, please switch AWS and GCP, and also put the second list in alphabetical order.

Author: Done. The second list I still put NFS and iSCSI first because essentially those are different than the rest since they are particular storage protocols compared to specific storage systems. The rest I am sorting to alphabetical

>     Automatically mount the storage system of your choice, whether from local storage, a public cloud provider such as <a href="https://aws.amazon.com/products/storage/">AWS</a> or <a href="https://cloud.google.com/storage/">GCP</a>, or a network storage system such as NFS, iSCSI, Ceph, Cinder, Gluster.

Reviewer 2: s/Gluster/GlusterFS

Events:                <none>
```

> When the `CSIMigration{provider}` feature flag is enabled for a specific in-tree volume plugin, the `kubernetes.io/pv-controller` finalizer is replaced by the `external-provisioner.volume.kubernetes.io/finalizer` finalizer.

Reviewer 3: Please wrap the long lines.

```suggestion
When the `CSIMigration{provider}` feature flag is enabled for a specific in-tree volume plugin,
the `kubernetes.io/pv-controller` finalizer is replaced by the
`external-provisioner.volume.kubernetes.io/finalizer` finalizer.
```

Author: Done

Events:                <none>
```

> `CSIMigration` feature is GA now, that means for a specific in-tree volume plugin that supports CSI migration and the
> specific `CSIMigration{provider}` feature flag is enabled, it will remove the `kubernetes.io/pv-controller` finalizer,
> while adding the `external-provisioner.volume.kubernetes.io/finalizer` finalizer.

Reviewer 4: ```suggestion
When the `CSIMigration{provider}` feature flag is enabled for a specific in-tree volume plugin, the `kubernetes.io/pv-controller` finalizer is replaced by the `external-provisioner.volume.kubernetes.io/finalizer` finalizer.
```

Author: Done

Events:                <none>
```

> Enabling the `CSIMigration` feature for a specific in-tree volume plugin will remove

Reviewer 5: Should we still keep the first half of this paragraph?

`CSIMigration` feature is GA now, that means for a specific in-tree volume plugin it will remove
the `kubernetes.io/pv-controller` finalizer, while adding the `external-provisioner.volume.kubernetes.io/finalizer`
finalizer.

  (**deprecated** in v1.18)
* [`flexVolume`](/docs/concepts/storage/volumes/#flexvolume) - FlexVolume
  (**deprecated** in v1.23)
* [`gcePersistentDisk`](/docs/concepts/storage/volumes/#gcepersistentdisk) - GCE Persistent Disk
  (**deprecated** in v1.17)
* [`glusterfs`](/docs/concepts/storage/volumes/#glusterfs) - Glusterfs volume
  (**deprecated** in v1.25)
* [`portworxVolume`](/docs/concepts/storage/volumes/#portworxvolume) - Portworx volume
  (**deprecated** in v1.25)
* [`vsphereVolume`](/docs/concepts/storage/volumes/#vspherevolume) - vSphere VMDK volume
  (**deprecated** in v1.19)

Older versions of Kubernetes also supported the following in-tree PersistentVolume types:

* `photonPersistentDisk` - Photon controller persistent disk.
  (**not available** after v1.15)
* [`scaleIO`](/docs/concepts/storage/volumes/#scaleio) - ScaleIO volume
  (**not available** after v1.21)
* [`flocker`](/docs/concepts/storage/volumes/#flocker) - Flocker storage
  (**not available** after v1.25)
* [`quobyte`](/docs/concepts/storage/volumes/#quobyte) - Quobyte volume
  (**not available** after v1.25)
* [`storageos`](/docs/concepts/storage/volumes/#storageos) - StorageOS volume

>   (**not available** after v1.25)

Reviewer 3: "after v1.25" means "starting v1.26" ?
Please be specific.

Author: it actually means starting 1.25. Good catch. Updated it
